SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SchemaGenerator.cpp

```diff
@@ -301,6 +301,9 @@ void GenerateSubobjectSchema(FComponentIdGenerator& IdGenerator, UClass* Class,
 
 	// Use previously generated component IDs when possible.
 	const FSubobjectSchemaData* const ExistingSchemaData = SubobjectClassPathToSchema.Find(Class->GetPathName());
+	checkf(ExistingSchemaData == nullptr || ExistingSchemaData->GeneratedSchemaName == ClassPathToSchemaName[Class->GetPathName()],
```

Comment:
Worried that this could fail in some circumstances considering how easy it is to break this code. Does this have to be a `check`? Maybe an error?